hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/messagevalidation/ExceptionHandler.java

```diff
@@ -38,24 +35,36 @@ public void process(Exchange exchange) {
 		Exception exception = exchange.getProperty(Exchange.EXCEPTION_CAUGHT, Exception.class);
 		
 		if (exception instanceof CustomHNSException) {
-			if (CustomError_Msg_InvalidAuthKey.getErrorMessage().equals(exception.getMessage())) {
-				exchange.getIn().setHeader(Exchange.HTTP_RESPONSE_CODE, SC_FORBIDDEN);
-			} else if (CustomError_Msg_InvalidRequest.getErrorMessage().equals(exception.getMessage())){
-				exchange.getIn().setHeader(Exchange.HTTP_RESPONSE_CODE, SC_BAD_REQUEST);
-			} else if (HL7Error_Msg_NoInputHL7.getErrorMessage().equals(exception.getMessage())){
-				exchange.getIn().setHeader(Exchange.HTTP_RESPONSE_CODE, SC_UNPROCESSABLE_ENTITY);
-			} else {
-				exchange.getIn().setHeader(Exchange.HTTP_RESPONSE_CODE, SC_BAD_REQUEST);
+			ErrorMessage errorMessage = ((CustomHNSException)exception).getErrorMessage(); 
+			switch (errorMessage) {
+			case HL7Error_Msg_NoInputHL7:
+			case CustomError_Msg_InvalidRequest:
+				generateErrorResponse(exchange, errorMessage, HttpStatus.SC_BAD_REQUEST);
+				break;
+			case CustomError_Msg_InvalidAuthKey:
+			case CustomError_Msg_MissingAuthKey:
+				generateErrorResponse(exchange, errorMessage, HttpStatus.SC_UNAUTHORIZED);
+				break;
+			default:
+				generateErrorResponse(exchange, ErrorMessage.CustomError_Msg_UknownError, HttpStatus.SC_BAD_REQUEST);
+				break;
 			}
 		} else if (exception instanceof HttpHostConnectException) {
-			exchange.getIn().setHeader(Exchange.HTTP_RESPONSE_CODE, SC_INTERNAL_SERVER_ERROR);
-			LOGGER.info("{} - Failed to connect remote server.",LoggingUtil.getMethodName());
+			logger.info("{} - Failed to connect remote server. {}", LoggingUtil.getMethodName(), exception.getMessage());
+			generateErrorResponse(exchange, ErrorMessage.CustomError_Msg_DownstreamConnectionFailed, HttpStatus.SC_INTERNAL_SERVER_ERROR);			
 		} else {
 			// Should not reach here as the specific exception should be handled above, add default error in case the specific handling not added
-			exchange.getIn().setHeader(Exchange.HTTP_RESPONSE_CODE, SC_INTERNAL_SERVER_ERROR);
+			generateErrorResponse(exchange, ErrorMessage.CustomError_Msg_UknownError, HttpStatus.SC_INTERNAL_SERVER_ERROR);
 		}
-		
-		//Set the body to null as none is expected.
-        exchange.getIn().setBody(null);
+	}
+	
+	private void generateErrorResponse(Exchange exchange, ErrorMessage errorMessage, int httpStatusCode) {
```

Comment:
httpStatusCode is never used, can that be removed?

Reply:
This is a bug. It should be used instead of the hardcoded 400.